analyzers/tests/SonarAnalyzer.UnitTest/Helpers/TypeHelperTest.cs

```diff
@@ -89,5 +89,173 @@ public void Type_GetSymbolType_Alias()
             symbol.ToString().Should().Be("PropertyBag");
             type.ToString().Should().Be("System.Collections.Generic.Dictionary<string, object>");
         }
+
+        [DataTestMethod]
+        [DataRow("int")]
+        [DataRow("System.Int32")]
+        [DataRow("int?")]
+        [DataRow("System.Nullable<int>")]
+        [DataRow("CustomStruct")]
+        [DataRow("CustomRefStruct")]
+        public void IsStruct_Simple(string type)
+        {
+            var (tree, model) = TestHelper.CompileCS($$"""
+                struct CustomStruct { }
+                ref struct CustomRefStruct { }
+
+                ref struct Test
+                {
+                    {{type}} field;
+                }
+                """);
+            var field = tree.GetRoot().DescendantNodes().OfType<FieldDeclarationSyntax>().First();
+            var fieldSymbol = (IFieldSymbol)model.GetDeclaredSymbol(field.Declaration.Variables[0]);
```

Comment:
Let's extract the `TestHelper.CompileCS` + finding syntax + finding `IFieldSymbol` into a method. Something like `IFieldSymbol CreateFieldSymbol(string code)`